pkg/registry/generic/registry/store.go

```diff
@@ -227,6 +230,22 @@ func (e *Store) Create(ctx api.Context, obj runtime.Object) (runtime.Object, err
 	if err := e.Storage.Create(ctx, key, obj, out, ttl); err != nil {
 		err = storeerr.InterpretCreateError(err, e.QualifiedResource, name)
 		err = rest.CheckGeneratedNameError(e.CreateStrategy, err, obj)
+		if !storage.IsNodeExist(err) && e.OnCreateFailure == nil {
+			return nil, err
+		}
+		if errGet := e.Storage.Get(ctx, key, out, false); errGet != nil {
+			return nil, storeerr.InterpretGetError(errGet, e.QualifiedResource, name)
+		}
+		accessor, errGetAcc := meta.Accessor(out)
+		if errGetAcc != nil {
+			return nil, kubeerr.NewInternalError(err)
+		}
+		if accessor.GetDeletionTimestamp() != nil {
+			err.(*kubeerr.StatusError).ErrStatus.Message += "- the object may be in the process of deletion!\n"
+		}
+		if e.OnCreateFailure != nil {
```

Comment:
Why do you have `OnCreateFailure` if you already append the message above?  Only need to do it in one place rather than two.  I'd prefer not to create a new HandlerFunc on Pod if you're already going to override it here.